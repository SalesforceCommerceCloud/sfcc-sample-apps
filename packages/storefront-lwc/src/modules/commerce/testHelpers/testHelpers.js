export function tick(time = null) {
    return new Promise(resolve => setTimeout(resolve, time));
}
