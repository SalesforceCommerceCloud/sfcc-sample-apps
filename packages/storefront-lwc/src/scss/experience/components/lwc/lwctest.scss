.lwctest {
    color: chocolate;
    background-color: azure;
}
