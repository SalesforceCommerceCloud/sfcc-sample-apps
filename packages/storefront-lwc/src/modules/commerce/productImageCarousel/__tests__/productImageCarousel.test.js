import { createElement } from 'lwc';
import ProductImageCarousel from 'commerce/productImageCarousel';

import { registerWireService } from '@lwc/wire-service';
import { register } from 'lwc';
import { tick } from 'commerce/testHelpers';

registerWireService(register);

describe('<commerce-product-image-carousel>', () => {
    afterEach(() => {
        // The jsdom instance is shared across test cases in a single file so reset the DOM
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('should render carousel', async () => {
        const element = createElement('commerce-product-image-carousel', {
            is: ProductImageCarousel,
        });
        element.images = mockImages;
        document.body.appendChild(element);

        await tick();

        // Initial carousel image
        expect(
            element.shadowRoot.querySelector('.carousel-item.active'),
        ).toMatchSnapshot();
    });

    it('should forward in the carousel', async () => {
        const element = createElement('commerce-product-image-carousel', {
            is: ProductImageCarousel,
        });
        element.images = mockImages;
        document.body.appendChild(element);

        await tick();

        // Initial carousel image
        expect(
            element.shadowRoot.querySelector('.carousel-item.active'),
        ).toMatchSnapshot();

        element.shadowRoot.querySelector('.carousel-control-next').click();

        await tick();

        // Next carousel image
        expect(
            element.shadowRoot.querySelector('.carousel-item.active'),
        ).toMatchSnapshot();

        element.shadowRoot.querySelector('.carousel-control-next').click();
        await tick();
        element.shadowRoot.querySelector('.carousel-control-next').click();
        await tick();

        // Back to the first carousel image
        expect(
            element.shadowRoot.querySelector('.carousel-item.active'),
        ).toMatchSnapshot();
    });

    it('should previous in the carousel', async () => {
        const element = createElement('commerce-product-image-carousel', {
            is: ProductImageCarousel,
        });
        element.images = mockImages;
        document.body.appendChild(element);

        await tick();

        // Initial carousel image
        expect(
            element.shadowRoot.querySelector('.carousel-item.active'),
        ).toMatchSnapshot();

        element.shadowRoot.querySelector('.carousel-control-prev').click();

        await tick();

        // Next carousel image
        expect(
            element.shadowRoot.querySelector('.carousel-item.active'),
        ).toMatchSnapshot();

        element.shadowRoot.querySelector('.carousel-control-prev').click();
        await tick();
        element.shadowRoot.querySelector('.carousel-control-prev').click();
        await tick();

        // Back to the first carousel image
        expect(
            element.shadowRoot.querySelector('.carousel-item.active'),
        ).toMatchSnapshot();
    });
});

const mockImages = [
    {
        title: 'Checked Silk Tie, ',
        alt: 'Checked Silk Tie, , large',
        link: 'https://image-1.jpg',
        __typename: 'Image',
    },
    {
        title: 'Checked Silk Tie, ',
        alt: 'Checked Silk Tie, , large',
        link: 'https://image-2.jpg',
        __typename: 'Image',
    },
    {
        title: 'Checked Silk Tie, Navy',
        alt: 'Checked Silk Tie, Navy, large',
        link: 'https://image-3.jpg',
        __typename: 'Image',
    },
];
