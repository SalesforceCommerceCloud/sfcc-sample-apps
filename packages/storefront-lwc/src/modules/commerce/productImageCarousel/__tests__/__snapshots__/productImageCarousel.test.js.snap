// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<commerce-product-image-carousel> should forward in the carousel 1`] = `
<div
  class="carousel-item active"
>
  <img
    class="d-block img-fluid"
    src="https://image-1.jpg"
  />
</div>
`;

exports[`<commerce-product-image-carousel> should forward in the carousel 2`] = `
<div
  class="carousel-item active"
>
  <img
    class="d-block img-fluid"
    src="https://image-2.jpg"
  />
</div>
`;

exports[`<commerce-product-image-carousel> should forward in the carousel 3`] = `
<div
  class="carousel-item active"
>
  <img
    class="d-block img-fluid"
    src="https://image-1.jpg"
  />
</div>
`;

exports[`<commerce-product-image-carousel> should previous in the carousel 1`] = `
<div
  class="carousel-item active"
>
  <img
    class="d-block img-fluid"
    src="https://image-1.jpg"
  />
</div>
`;

exports[`<commerce-product-image-carousel> should previous in the carousel 2`] = `
<div
  class="carousel-item active"
>
  <img
    class="d-block img-fluid"
    src="https://image-3.jpg"
  />
</div>
`;

exports[`<commerce-product-image-carousel> should previous in the carousel 3`] = `
<div
  class="carousel-item active"
>
  <img
    class="d-block img-fluid"
    src="https://image-1.jpg"
  />
</div>
`;

exports[`<commerce-product-image-carousel> should render carousel 1`] = `
<div
  class="carousel-item active"
>
  <img
    class="d-block img-fluid"
    src="https://image-1.jpg"
  />
</div>
`;
